modules/apps/object/object-api/src/main/java/com/liferay/object/exception/RequiredObjectFieldException.java

```diff
@@ -18,10 +19,83 @@ public RequiredObjectFieldException() {
 		_messageKey = "at-least-one-object-field-must-be-added";
 	}
 
+	public RequiredObjectFieldException(String message, String messageKey) {
+		super(message);
+
+		_messageKey = messageKey;
+	}
+
 	public String getMessageKey() {
 		return _messageKey;
 	}
 
+	public static class MustHaveFiveObjectFieldsAtMaximum
+		extends RequiredObjectFieldException {
+
+		public MustHaveFiveObjectFieldsAtMaximum() {
+			super(
+				"Add a maximum of five object fields to create unique " +
+					"composite keys",
+				"add-a-maximum-of-five-object-fields-to-create-unique-" +
+					"composite-keys");
+		}
+
+	}
+
+	public static class MustHaveTwoObjectFieldsAtMinimum
+		extends RequiredObjectFieldException {
+
+		public MustHaveTwoObjectFieldsAtMinimum() {
+			super(
+				"Add a minimum of two fields to create unique composite keys",
+				"add-a-minimum-of-two-fields-to-create-unique-composite-keys");
+		}
+
+	}
+
+	public static class MustNotDeleteObjectFieldPublishedObjectDefinition
+		extends RequiredObjectFieldException {
+
+		public MustNotDeleteObjectFieldPublishedObjectDefinition() {
+			super(
+				"Fields cannot be deleted from unique composite keys after " +
+					"the definition is published",
+				"fields-cannot-be-deleted-from-unique-composite-keys-after-" +
+					"the-definition-is-published");
+		}
+
+	}
+
+	public static class MustNotDeleteObjectFieldUniqueCompositeKey
+		extends RequiredObjectFieldException {
+
+		public MustNotDeleteObjectFieldUniqueCompositeKey() {
+			super(
+				StringBundler.concat(
+					"This object field cannot be deleted as it is used in a ",
+					"Composite Unique Key Validation. To remove this object ",
+					"field you must first delete the associated Unique ",
+					"Composite Key Validation."),
+				StringBundler.concat(
+					"this-object-field-cannot-be-deleted-as-it-is-used-in-a-",
+					"composite-unique-key-validation.-to-remove-this-object-",
+					"field-you-must-first-delete-the-associated-unique-",
+					"composite-key-validation"));
+		}
+
+	}
+
+	public static class MustNotUseFilledObjectField
+		extends RequiredObjectFieldException {
+
+		public MustNotUseFilledObjectField() {
+			super(
+				"The field already have values tied to it",
+				"The-field-already-have-values-tied-to-it");
+		}
+
+	}
```

Comment:
@mateussgn The `RequiredObjectFieldException` should only be used when trying to delete the object field and it is not possible, which means the object field is **required** in the **database**.

Please reuse the existing `ObjectValidationRuleSettingValueException.InvalidValue` as this exception is related to the value the user is trying to add in the `ObjectValidationRuleSetting` table.